pkg/deployer/manifest/ensure.go

```diff
@@ -64,15 +68,53 @@ func (m *Manifest) Reconcile(ctx context.Context) error {
 
 	m.DeployItem.Status.ProviderStatus, err = encodeStatus(m.ProviderStatus)
 	if err != nil {
-		m.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(m.DeployItem.Status.LastError,
+		return lsv1alpha1helper.NewWrappedError(err,
 			currOp, "ProviderStatus", err.Error())
-		return err
+	}
+	if err := m.kubeClient.Status().Update(ctx, m.DeployItem); err != nil {
+		return lsv1alpha1helper.NewWrappedError(err,
+			currOp, "UpdateStatus", err.Error())
+	}
+
+	if m.ProviderConfiguration.HealthChecks.DisableDefault {
+		m.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
+		m.DeployItem.Status.ObservedGeneration = m.DeployItem.Generation
+		m.DeployItem.Status.LastError = nil
+		return nil
+	}
+
+	return m.CheckResourcesHealth(ctx, targetClient)
```

Comment:
same as in the helm deployer

Reply:
Moved the code into the `CheckResourcesHealth` function and delegated most of that code into a `defaultCheckResourcesHealth` function to already provide the prerequisites to implement custom health checks.